Source/Core/CartographicGeocoderService.js

```diff
@@ -0,0 +1,63 @@
+/*global define*/
+define([
+    './Cartesian3',
+    './defaultValue',
+    './defineProperties',
+    './defined',
+    './DeveloperError',
+    '../ThirdParty/when'
+], function(
+    Cartesian3,
+    defaultValue,
+    defineProperties,
+    defined,
+    DeveloperError,
+    when) {
+    'use strict';
+
+    /**
+     * Geocodes queries containing longitude and latitude coordinates and an optional height.
+     * Query format: `longitude latitude (height)` with longitude/latitude in degrees and height in meters.
+     *
+     * @alias CartographicGeocoderService
+     * @constructor
+     */
+    function CartographicGeocoderService() {
+    }
+
+    /**
+     * @function
+     *
+     * @param {String} query The query to be sent to the geocoder service
+     * @returns {Promise<GeocoderResult[]>}
+     */
+    CartographicGeocoderService.prototype.geocode = function(query) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(query)) {
+            throw new DeveloperError('query must be defined');
+        }
+        //>>includeEnd('debug');
+
+        try {
```

Comment:
Sorry if this has been covered, but what part of this try block can cause an exception?